lambdas/cmr-query/handler.py

```diff
@@ -1,11 +1,102 @@
+        data_file: str
+            string value describing the data file from which to build the STAC item
+        data_file_regex: str
+            string value becomes a regex pattern to find all related data file urls,
+            commonly a product ID or other identifier shared amongst product files
+        data: List[Dict[str, Any]]
+            dictionary of file_obj's generated from querying CMR
+    Return:
+        objects: List[Dict[str, Any]]
+            modified dictionary of passed in file_obj's, used to generate STAC items
+
+    Example:
+        multi_asset_items(
+            "cov_1-1.hdr".
+            "uavsar_AfriSAR_v1-.*_\d{5}_\d{5}_\d{3}_\d{3}_\d{6}",
+            {
+                'collection': 'AfriSAR_UAVSAR_Ungeocoded_Covariance',
+                'remote_fileurl': 's3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_Ungeocoded_Covariance___1/uavsar_AfriSAR_v1-cov_topo_a41_r9_localPsi.rdr',
+                'granule_id': 'G1200110696-NASA_MAAP',
+                'id': 'G1200110696-NASA_MAAP',
+                'mode': 'cmr',
+                'test_links': None,
+                'reverse_coords': None,
+                'asset_name': 'data',
+                'asset_roles': ['data'],
+                'asset_media_type': 'application/x-hdr'
+            }
+        )
+        Returns:
+        {
+            'collection': 'AfriSAR_UAVSAR_Ungeocoded_Covariance',
+            'remote_fileurl': 's3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_Ungeocoded_Covariance___1/uavsar_AfriSAR_v1-cov_coreg_fine_hsixty_14050_16015_140_009_160308_cov_1-1.hdr',
+            'granule_id': 'G1200109928-NASA_MAAP',
+            'id': 'G1200109928-NASA_MAAP',
+            'mode': 'cmr',
+            'test_links': None,
+            'reverse_coords': None,
+            'asset_name': 'data',
+            'asset_roles': ['data'],
+            'asset_media_type': 'application/x-hdr',
+            'assets': {
+                'cov_1-1.bin': <Asset href=s3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_Ungeocoded_Covariance___1/uavsar_AfriSAR_v1-cov_coreg_fine_hsixty_14050_16015_140_009_160308_cov_1-1.bin>,
+                'cov_1-1.hdr': <Asset href=s3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_Ungeocoded_Covariance___1/uavsar_AfriSAR_v1-cov_coreg_fine_hsixty_14050_16015_140_009_160308_cov_1-1.hdr>,
+                'cov_1-2.bin': <Asset href=s3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_Ungeocoded_Covariance___1/uavsar_AfriSAR_v1-cov_coreg_fine_hsixty_14050_16015_140_009_160308_cov_1-2.bin>,
+                ...,
+            }
+            'product_id': 'uavsar_AfriSAR_v1-cov_coreg_fine_hsixty_14050_16015_140_009_160308'
+        }
+    """
+    fileurls_pattern = re.compile(data_file_regex)
+    objects = []
+    product_ids = {}
+
+    def _get_asset_name(remote_fileurl: str, product_id: str) -> str:
+        return re.sub(f".*{product_id}[-_]?", "", remote_fileurl)
```

Comment:
is this just particular to the AfriSAR collections or you can expect this format in the file url consistently?

Reply:
Right, I had noted that in the PR comments at the top - I _believe_ this to be the common filename format but of course there are exceptions even within the AfriSAR collections